app/src/main/java/com/g/tragosapp/ui/cocktaildetails/CocktailsDetailFragment.kt

```diff
@@ -46,11 +52,64 @@ class CocktailsDetailFragment : Fragment() {
         binding.cocktailDesc.text = cocktail.description
 
-        binding.btnSaveCocktail.setOnClickListener {
-            viewModel.saveCocktail(FavoritesEntity(cocktail.cocktailId,cocktail.image,cocktail.name,cocktail.description,cocktail.hasAlcohol))
-            Toast.makeText(requireContext(), "Cocktail saved to favorites", Toast.LENGTH_SHORT)
-                .show()
-        }
+        viewModel.getFavoriteCocktails().observe(viewLifecycleOwner, Observer { result ->
+            when (result) {
+                is Resource.Loading -> {
+                }
+                is Resource.Success -> {
+                    when (result.data.contains(cocktail)) {
+                        true -> {
+                            binding.btnSaveCocktail.setOnClickListener {
+                                viewModel.deleteCocktail(
+                                    FavoritesEntity(
+                                        cocktail.cocktailId,
+                                        cocktail.image,
+                                        cocktail.name,
+                                        cocktail.description,
+                                        cocktail.hasAlcohol
+                                    )
+                                ).observe(viewLifecycleOwner, Observer {
+                                    if (it is Resource.Success) {
+                                        context?.shortToast("Drink deleted !")
+                                        findNavController().navigate(
+                                            R.id.action_tragosDetalleFragment_self,
+                                            Bundle().apply { putParcelable("drink", cocktail) }
+                                        )
+                                    }
+                                })
+                            }
+                            binding.btnSaveCocktail.setImageDrawable(
+                                getDrawable(requireContext(), R.drawable.ic_baseline_favorite_24)
+                            )
+                        }
+                        false -> {
+                            binding.btnSaveCocktail.setOnClickListener {
+                                viewModel.saveCocktail(
+                                    FavoritesEntity(
+                                        cocktail.cocktailId,
+                                        cocktail.image,
+                                        cocktail.name,
+                                        cocktail.description,
+                                        cocktail.hasAlcohol
+                                    )
+                                )
+                                context?.shortToast("Cocktail saved to favorites")
+                                findNavController().navigate(
+                                    R.id.action_tragosDetalleFragment_self,
+                                    Bundle().apply { putParcelable("drink", cocktail) }
```

Comment:
Is necessary to reload the fragment after saving the cocktail? I think the line below that changes the button drawable should be enought

Reply:
NECESSARY! because we need change ICON STUPID!!

Reply:
I'm not arguing with you, I'm just asking a question, there is no need of reloading again the same fragment to change a button icon, you can do it without refreshing the whole view, its just a button

Reply:
Why are you calling people stupid for not immediately agreeing with you? 🤔